dali/operators/decoder/nvjpeg/nvjpeg_memory.cc

```diff
@@ -139,7 +139,8 @@ struct NVJpegMem {
     auto ptr = mm::alloc_raw_unique<char>(mr, size);
     std::unique_lock<std::shared_timed_mutex> lock(alloc_info_mutex_);
     AllocInfo &ai = alloc_info_[ptr.get()];  // create entry first (before moving out the deleter)
-    ai = { mm::memory_kind_id::device, std::move(ptr.get_deleter()), size, thread_id};
```

Comment:
This caused host memory to be returned as device memory and then we could pick it up as such. It worked with pinned memory, because it is device accessible (albeit slower).